platform/ext/target/nordic_nrf/common/core/tfm_hal_its_encryption.c

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "platform/include/tfm_platform_system.h"
+#include "platform/include/tfm_hal_its.h"
+#include "psa/crypto.h"
+
+#include "nrf_cc3xx_platform_derived_key.h"
+#include "nrf_cc3xx_platform.h"
+#include "nrf_cc3xx_platform_kmu.h"
+#include <string.h>
+
+#define ITS_ENCRYPTION_SUCCESS 0
+
+#define HUK_KMU_SLOT 2
+#define HUK_KMU_SIZE_BITS 128
+
+/* Global encryption counter which resets per boot. The counter ensures that
+ * the nonce will not be identical for consecutive file writes during the same
+ * boot.
+ */
+static uint32_t g_enc_counter;
+
+/* The global nonce seed which is fetched once in every boot. The seed is used
+ * as part of the nonce and allows the platforms to diversify their nonces
+ * across resets. Note that the way that this seed is generated is platform
+ * specific, so the diversification is optional.
+ */
+static uint8_t g_enc_nonce_seed[TFM_ITS_ENC_NONCE_LENGTH -
+                                sizeof(g_enc_counter)];
```

Comment:
```suggestion
static uint8_t g_enc_nonce_seed[8];
```

BUILD_ASSERT(TFM_ITS_ENC_NONCE_LENGTH == 12, "This implementation only supports a nonce of size 12");